### PR TITLE
feat: be explicit when specifying time & replace moment with date-fns

### PR DESCRIPTION
**Pull request by theneva**
This uses `date-fns` for all (I hope) fiddling with time and durations.

I think this makes it way easier to understand which units are used (for example, milliseconds or seconds). It also removes the need for things like hard coded millisecond representations of 24 hours.

I took the liberty of replacing all usage of `moment` with `date-fns`, and removing the dependency. (I did not, however, update the docs page that uses moment in an example (website/docs/advanced/custom-activation-strategy.md)).

I'll write some inline PR comments in the places where I think the changes are less than obvious – but feel free to ask questions :smile:

A small note: I've not been super strict about using just one of `Date.now()` and `new Date()`, since it doesn't matter. `date-fns` happily consumes both.

Do you think this makes sense?

### CODE REVIEW

File: src/lib/addons/addon.test.ts [hunk 1]
```diff
@@ -19,7 +20,7 @@ test('Retries if fetch throws', async () => {
         statusText: 'ACCEPTED',
     });
     await addon.fetchRetry(url);
-    jest.advanceTimersByTime(1000);
+    jest.advanceTimersByTime(millisecondsInSecond);
```
[the remark this answers is not in the document]
Comment (theneva) [reply src/lib/addons/addon.test.ts]: (I did it like that :smile:)

File: src/lib/services/client-metrics/client-metrics.test.ts [hunk 2]
```diff
@@ -252,11 +295,11 @@ test('should have correct values for lastMinute', () => {
     let c = metrics.getTogglesMetrics();
     expect(c.lastMinute.toggle).toEqual({ yes: 20, no: 20 });
 
-    jest.advanceTimersByTime(10 * 1000);
+    jest.advanceTimersByTime(10_000);
```
Comment (theneva) [inline src/lib/services/client-metrics/client-metrics.test.ts:298]: yep, having played with it a bit, I think anything below 1 minute is better written as a constant with underscores than either `10 * 1000` or `secondsToMilliseconds(10)`

File: src/lib/services/client-metrics/client-metrics.test.ts [hunk 3]
```diff
@@ -318,27 +361,27 @@ test('should have correct values for lastHour', () => {
...
     expect(c.lastHour.toggle).toEqual({ yes: 41, no: 41 });
 
-    jest.advanceTimersByTime(10 * 1000);
+    jest.advanceTimersByTime(10_000);
     c = metrics.getTogglesMetrics();
     expect(c.lastHour.toggle).toEqual({ yes: 41, no: 41 });
 
     // at 30
-    jest.advanceTimersByTime(30 * 60 * 1000);
+    jest.advanceTimersByTime(minutesToMilliseconds(30));
```
Comment (theneva) [inline src/lib/services/client-metrics/client-metrics.test.ts:369]: … but when we're dealing with minutes, the numbers are less easy to grok, so I prefer the helper functions over either a constant number of milliseconds or inline arithmetic

File: src/lib/services/client-metrics/ttl-list.js [hunk 4]
```diff
@@ -36,26 +36,22 @@ module.exports = class TTLList extends EventEmitter {
...
-            now.isBefore(value.ttl),
-        );
+        this.list.reverseRemoveUntilTrue(({ value }) => isFuture(value.ttl));
         this.startTimer();
     }
 
     destroy() {
-        // https://github.com/nodejs/node/issues/9561
-        // clearTimeout(this.timer);
-        // this.timer = null;
```
[the remark this answers is not in the document]
Comment (theneva) [reply src/lib/services/client-metrics/ttl-list.js:58]: okay, I'll leave my original change in, then :smile:

File: src/lib/services/client-metrics/ttl-list.test.js [hunk 5]
```diff
@@ -43,18 +43,45 @@ test('should slice off list', () => {
...
 
     jest.advanceTimersByTime(301);
     expect(expired).toHaveLength(4);
+    expect(list.list.toArray()).toHaveLength(0);
 
     list.destroy();
     jest.useRealTimers();
 });
+
+test('should add item created in the past but expiring in the future', () => {
```
Comment (theneva) [inline src/lib/services/client-metrics/ttl-list.test.js:69]: I added this to ensure it both works with milliseconds and to catch a bug that I almost introduced and the other tests didn't catch

**Comment by theneva** [comment]
Okay, I think this is ready for a final look :smile:

File: src/test/e2e/services/user-service.e2e.test.ts [hunk 6]
```diff
@@ -141,8 +142,8 @@ test("deleting a user should delete the user's sessions", async () => {
         sid: 'xyz321',
         sess: {
             cookie: {
-                originalMaxAge: 2880000,
-                expires: new Date(Date.now() + 86400000).toDateString(),
+                originalMaxAge: minutesToMilliseconds(48),
```
[the remark this answers is not in the document]
Comment (ivarconr) [reply src/test/e2e/services/user-service.e2e.test.ts:165]: I do not think that's needed.

**Review by ivarconr — APPROVED** [review]
LGTM!

**Pull request merged**